R/ttestindependentsamples.R

```diff
@@ -426,8 +439,15 @@ ttestIndependentMainTableRow <- function(variable, dataset, test, testStat, effS
   }
 }
 
-.ttestIndependentEqVarRow <- function(table, variable, groups, dataset) {
-  levene <- car::leveneTest(dataset[[ .v(variable) ]], dataset[[ .v(groups) ]], "mean")
+.ttestIndependentEqVarRow <- function(table, variable, groups, dataset, options) {
+
+  if (options$equalityOfVarianceType == "brownForsythe") {
+    levene <- car::leveneTest(dataset[[ .v(variable) ]], dataset[[ .v(groups) ]], "median")
+  }
+  else if (options$equalityOfVarianceType == "levene") {
+    levene <- car::leveneTest(dataset[[ .v(variable) ]], dataset[[ .v(groups) ]], "mean")
+  }
```

Comment:
Here I would again opt for just switch between "median" and "mean" instead of adding the switch on the whole leveneTest

```r
center <- if(options$equalityOfVarianceType == "brownForsythe") "median" else "mean"
levene <- car::leveneTest(dataset[[ .v(variable) ]], dataset[[ .v(groups) ]], center)
```